airflow/models.py

```diff
@@ -3444,7 +3458,7 @@ def find(dag_id=None, run_id=None, execution_date=None,
         return dr
 
     @provide_session
-    def get_task_instances(self, state=None, session=None):
+    def get_task_instances(self, state=None, to_list=True, session=None):
```

Comment:
Actually, the only place that I see this used is in jobs.py (twice), and in tests. Would prefer removing the to_list param, and just returning tis. Then updating code that makes the call to use .all() on the return object.

Ack that this is backwards compatible, but seems very limited usage.